src/test_l3_dsDisplay.c

```diff
@@ -0,0 +1,435 @@
+             edid->productCode, edid->serialNumber, edid->manufactureYear, edid->manufactureWeek, edid->hdmiDeviceType, edid->isRepeater, edid->physicalAddressA, edid->physicalAddressB, edid->physicalAddressC, edid->physicalAddressD, edid->numOfSupportedResolution, edid->monitorName);
+    DS_ASSERT(status == dsERR_NONE);
+
+    free(edid);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+ * @brief Retrieves the Extended Display Identification Data (EDID) bytes for the display.
+ *
+ * Retrieves the EDID data bytes for the display associated with the selected handle.
+ *
+ * **Test Group ID:** 03@n
+ * **Test Case ID:** 005@n
+ *
+ * **Test Procedure:**
+ * Refer to Test specification documentation
+ * [ds-display_L3_Low-Level_TestSpecification.md](../../docs/pages/ds-display_L3_Low-Level_TestSpecification.md)
+ */
+void dsDisplay_GetEDIDBytes()
+{
+    gTestID = 5;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    int length = 0;
+    unsigned char *edid = NULL;
+
+    dsGetDisplay_getHandle();
+
+    UT_LOG_INFO("Calling dsGetEDID(Handle:[0x%0X]])", displayHandle);
+    status = dsGetEDIDBytes(displayHandle, edid, &length);
+
+    UT_LOG_INFO("Result dsGetEDIDBytes()  dsError_t=[%s]", UT_Control_GetMapString(errorMappingTable, status));
+    DS_ASSERT(status == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+
+}
+
+/**
+ * @brief Retrieves the aspect ratio of the display.
+ *
+ * Retrieves the aspect ratio of the display.
+ *
+ * **Test Group ID:** 03@n
+ * **Test Case ID:** 006@n
+ *
+ * **Test Procedure:**
+ * Refer to Test specification documentation
+ * [ds-display_L3_Low-Level_TestSpecification.md](../../docs/pages/ds-display_L3_Low-Level_TestSpecification.md)
+ */
+void GetDisplay_AspectRatio()
+{
+    gTestID = 6;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status   = dsERR_NONE;
+    dsVideoAspectRatio_t displayAspectRatio;
+  
+    dsGetDisplay_getHandle();
+
+    UT_LOG_INFO("Calling dsGetEDID(IN:handle:[0x%0X]) ", displayHandle);
+    status = dsGetDisplayAspectRatio(displayHandle, &displayAspectRatio);
+
+    UT_LOG_INFO("Result dsGetDisplayAspectRatio(dsVideoAspectRatio_t:[%s], dsError_t=[%s])", UT_Control_GetMapString(dsVideoAspect_RatioMappingTable, displayAspectRatio), UT_Control_GetMapString(errorMappingTable, status));
+
+    DS_ASSERT(status == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+
+}
+
+/**
+ * @brief This test registers a callback function for display events.
+ *
+ * Registers a callback function to handle display events associated with the display handle.
+ *
+ * **Test Group ID:** 03@n
+ * **Test Case ID:** 007@n
+ *
+ * **Test Procedure:**
+ * Refer to Test specification documentation
+ * [ds-display_L3_Low-Level_TestSpecification.md](../../docs/pages/ds-display_L3_Low-Level_TestSpecification.md)
+ */
+void RegisterDisplayEventCallback()
```

Comment:
no need of extra function for call back registration. Move callback registration to dsGetDisplay_getHandle and add dsGetDisplay_getHandle to the menu.

Reply:
Done